q2_chemistree/_match.py

```diff
@@ -19,22 +19,29 @@ def match_label(collated_fingerprints: pd.DataFrame,
     binary fingerprint vector.
     '''
     fps = collated_fingerprints.copy()
-    allfps = set(fps.index)
+    allfps = list(fps.index)
     if fps.empty:
         raise ValueError("Cannot have empty fingerprint table")
     table = feature_table.to_dataframe(dense=True)
     allfeatrs = set(table.index)
-    if not allfps.issubset(allfeatrs):
-        extra_tips = allfps-allfps.intersection(allfeatrs)
+    if not set(allfps).issubset(allfeatrs):
+        extra_tips = set(allfps) - set(allfps).intersection(allfeatrs)
         raise ValueError('The following tips were not '
                          'found in the feature table:\n' +
                          ', '.join([str(i) for i in extra_tips]))
     filtered_table = table.reindex(allfps)
+    fps = fps.apply(pd.to_numeric)
```

Comment:
At this point are fingerprints saved as strings, or why do they need to be converted into numbers?

Reply:
I _think_ they are internally being cast as strings even though the code explicitly does not do this. I noticed that all fingerprints were equal to 1 if we do not explicitly change the table to numeric.

Reply:
That's rather odd, I believe BIOM should only be storing float data, maybe we should look into this as it might be a symptom of something else.